src/datadog_handler.cpp

```diff
@@ -53,7 +54,8 @@ ngx_int_t on_enter_block(ngx_http_request_t *request) noexcept try {
   } else {
     try {
       context->on_change_block(request, core_loc_conf, loc_conf);
-    } catch (...) {
+    } catch (const std::exception &e) {
+      telemetry::report_error_log(e.what(), CURRENT_FRAME(request));
```

Comment:
Using directly e.what() are we sure we are not taking any risks or leaking PII ?

Reply:
PHP is in the same case as you so we are not handling them (and I will have to check this is safe). 
In python i'm ensuring the log message is constant. 

If you know that we are not throwing exceptions with PIIs we don't have to make any changes.